client/src/components/UserManagement.jsx

```diff
@@ -0,0 +1,201 @@
+import * as React from "react";
+import Box from "@mui/material/Box";
+import { DataGrid, GridActionsCellItem, gridClasses } from "@mui/x-data-grid";
+import Button from "@mui/material/Button";
+import AddIcon from "@mui/icons-material/Add";
+import EditIcon from "@mui/icons-material/Edit";
+import DeleteIcon from "@mui/icons-material/DeleteOutlined";
+import Card from "@mui/material/Card";
+import CardActions from "@mui/material/CardActions";
+import CardContent from "@mui/material/CardContent";
+import TextField from "@mui/material/TextField";
+import InputAdornment from "@mui/material/InputAdornment";
+import SearchIcon from "@mui/icons-material/Search";
+import ArrowBackIcon from "@mui/icons-material/ArrowBack";
+import Typography from "@mui/material/Typography";
+
+const columns = [
+  {
+    field: "name",
+    headerName: "Name",
+    width: 535,
```

Comment:
Instead of hard coding a column width, we can set a `minWidth` which should help condense the size of the grid on small screens and prevent the need to sideways scroll
This applies to all the columns: name, email, actions